sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/tracing/LoggingTracerProvider.java

```diff
@@ -0,0 +1,190 @@
+ */
+public class LoggingTracerProvider implements TracerProvider {
+    private static final TracingOptions DEFAULT_OPTIONS = new TracingOptions().setEnabled(false);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Tracer createTracer(String libraryName, String libraryVersion, String azNamespace, TracingOptions options) {
+        return new LoggingTracer(options instanceof LoggingTracingOptions ? options : DEFAULT_OPTIONS);
+    }
+
+    /**
+     * Options to configure {@link LoggingTracerProvider}.
+     */
+    public static class LoggingTracingOptions extends TracingOptions {
+        /**
+         * Creates new instance of {@link LoggingTracingOptions}
+         */
+        public LoggingTracingOptions() {
+            super(LoggingTracerProvider.class);
+        }
+    }
+
+    /**
+     * Logging tracer implementation.
+     */
+    private static class LoggingTracer implements Tracer {
+        private final boolean isEnabled;
+        LoggingTracer(TracingOptions options) {
+            this.isEnabled = options.isEnabled();
+        }
+
+        @Override
+        public boolean isEnabled() {
+            return isEnabled;
+        }
+
+        @Override
+        public Context start(String name, Context context) {
+            if (!isEnabled) {
+                return context;
+            }
+            LoggingSpan span = new LoggingSpan(name, SpanKind.INTERNAL, getSpan(context));
+            return context.addData("span", span);
+        }
+
+        @Override
+        public Context start(String name, StartSpanOptions options, Context context) {
+            if (!isEnabled) {
+                return context;
+            }
+
+            LoggingSpan span = new LoggingSpan(name, options.getSpanKind(), getSpan(context));
+            options.getAttributes().forEach((k, v) -> span.addKeyValue(k, v));
+            span.addKeyValue("startTimestamp", options.getStartTimestamp());
+            return context.addData("span", span);
+        }
+
+        @Override
+        public void end(String errorMessage, Throwable throwable, Context context) {
+            if (isEnabled) {
+                LoggingSpan span = getSpan(context);
+                span.addKeyValue("status", errorMessage);
+                span.end(throwable);
+            }
+        }
+
+        @Override
+        public void setAttribute(String key, String value, Context context) {
+            getSpan(context).addKeyValue(key, value);
+        }
+
+        @Override
+        public void setAttribute(String key, long value, Context context) {
+            getSpan(context).addKeyValue(key, value);
+        }
+
+        @Override
+        public void injectContext(BiConsumer<String, String> headerSetter, Context context) {
+            LoggingSpan span = getSpan(context);
+            headerSetter.accept("traceparent", String.format("00-%s-%s-01", span.getTraceId(), span.getSpanId()));
```

Comment:
Is this something that would be called a lot? `String.format` has a fair bit of overhead compared to String concatenation.

Reply:
Good point!

This is only called from ServiceBus integration tests and I'd probably enable it in a few other integration tests. I don't expect any real-world usage unless we'll decide to enable log correlation when tracing is disabled. 

If there will be any interest I will definitely optimize, test, and move it out of the experimental package.